### DISCUSSION TITLE
Literature reference for generalized proximal gradient

### DISCUSSION THREAD

**Discussion by ner0-m** [PyLops/pyproximal#128] [Q&A]
Hi!

I am interested in the [GeneralizedProximalGradient](https://pyproximal.readthedocs.io/en/stable/api/generated/pyproximal.optimization.primal.GeneralizedProximalGradient.html) algorithm implemented in PyProximal (As I think the extension for multiple prox friendly functions is very interesting).

However, there is no mention of a reference/literature in the docs or source code as far as I can tell. Does anyone know a reference for this algorithm?  I would highly appreciate it!

**Comment by mrava87** [comment]
Sorry @ner0-m, I missed this. I tag @olivierleblanc who implemented the [GeneralizedProximalGradient](https://pyproximal.readthedocs.io/en/stable/api/generated/pyproximal.optimization.primal.GeneralizedProximalGradient.html)

Indeed, it would be good to add a reference to the Notes in the docstring 😄

**Reply by olivierleblanc** [reply]
Hi guys,
Here is the reference: https://arxiv.org/abs/1108.4404
@mrava87 can you add this reference to the Notes?

**Comment by mrava87** [comment]
Thanks! Done https://github.com/PyLops/pyproximal/pull/243
